opennms-services/src/main/java/org/opennms/netmgt/collectd/SnmpCollectionSetRefactorDummy.java

```diff
@@ -0,0 +1,105 @@
+ * in order to refactor the SnmpCollertor to use the CollectionsetBuilder.
+ */
+public class SnmpCollectionSetRefactorDummy extends SnmpCollectionSet implements Collectable, CollectionSet {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SnmpCollectionSetRefactorDummy.class);
+
+    SnmpCollectionSetRefactorDummy(SnmpCollectionAgent agent, OnmsSnmpCollection snmpCollection, LocationAwareSnmpClient client) {
+        super(agent,  snmpCollection, client);
+    }
+
+    CollectionSet createCollectionSet() {
+        CollectionSetBuilder builder = new CollectionSetBuilder(getAgent());
+
+
+        // TODO Patrick Very ugly mapping until refactoring is done. We convert here the snmp world into the CollectionSetBuilder world
+        for (CollectionResource collectionResource : getResources()) {
+            if(collectionResource instanceof NodeInfo) {
+                NodeInfo nodeInfo = (NodeInfo) collectionResource;
+                NodeLevelResource resource = new NodeLevelResource(nodeInfo.getNodeId());
+                addGroupsToBuilder(builder, resource, nodeInfo.getGroups());
+            }
+
+            if(collectionResource instanceof IfInfo) {
+                IfInfo ifInfo = (IfInfo) collectionResource;
+                InterfaceLevelResource resource = new InterfaceLevelResource(new NodeLevelResource(ifInfo.getNodeId()), ifInfo.getAttributesMap().get("snmpifname"));
+                addGroupsToBuilder(builder, resource, ifInfo.getGroups());
+            }
+
+            if(collectionResource instanceof AliasedResource) {
+                AliasedResource aliasedResource = (AliasedResource) collectionResource;
+                Resource resource = null; // TODO: Patrick there is no AliasedResource in the CollectionSetBuilder world, what shall we do?
+                addGroupsToBuilder(builder, resource, aliasedResource.getGroups());
+            }
+
+            if(collectionResource instanceof GenericIndexResource) {
+                GenericIndexResource genericResource = (GenericIndexResource) collectionResource;
+                org.opennms.netmgt.config.datacollection.ResourceType resourceType = new org.opennms.netmgt.config.datacollection.ResourceType();
+                resourceType.setName(genericResource.getResourceTypeName());
+                resourceType.setLabel(genericResource.getResourceTypeName());
+                // TODO Patrick what shall we set here? resourceType.setPersistenceSelectorStrategy();
+                // TODO Patrick what shall we set here?  resourceType.setResourceLabel();
+                // TODO Patrick what shall we set here?  resourceType.setStorageStrategy(nodeInfo.getResourceType().);
+                GenericTypeResource resource = new GenericTypeResource(new NodeLevelResource(genericResource.getCollectionAgent().getNodeId()),resourceType,genericResource.getInstance());
```

Comment:
Consider using the `DeferredGenericTypeResource` here if you don't have the actual type definition available.

Reply:
:+1:

Reply:
Sounds good. I tried but then it fails in DeferredGenericTypeResource.resolve since there is no ResourceTypeMapper...